images/virtualization-artifact/pkg/controller/cvi/internal/inuse.go

```diff
@@ -0,0 +1,333 @@
+		if vi.Status.Phase != virtv2.ImageReady && vi.Status.Phase != virtv2.ImageTerminating {
+			visNotReady = append(visNotReady, &vi)
+		}
+	}
+
+	return visNotReady, nil
+}
+
+func (h InUseHandler) listCVIsUsingImage(ctx context.Context, cvi *virtv2.ClusterVirtualImage) ([]client.Object, error) {
+	var cvis virtv2.ClusterVirtualImageList
+	err := h.client.List(ctx, &cvis, client.MatchingFields{
+		indexer.IndexFieldCVIByCVIDataSource: cvi.GetName(),
+	})
+	if err != nil {
+		return []client.Object{}, err
+	}
+
+	var cvisNotReady []client.Object
+	for _, cviItem := range cvis.Items {
+		if cviItem.Status.Phase != virtv2.ImageReady && cviItem.Status.Phase != virtv2.ImageTerminating {
+			cvisNotReady = append(cvisNotReady, &cviItem)
+		}
+	}
+
+	return cvisNotReady, nil
+}
+
+func (h InUseHandler) listNamespacesUsingImage(vms []client.Object, vmbdas []client.Object, vds []client.Object, vis []client.Object) []string {
+	var objects []client.Object
+	objects = append(objects, vms...)
+	objects = append(objects, vmbdas...)
+	objects = append(objects, vds...)
+	objects = append(objects, vis...)
+
+	namespacesMap := make(map[string]struct{})
+	for _, obj := range objects {
+		namespace := obj.GetNamespace()
+		if namespace == "" {
+			namespace = "default"
+		}
+
+		namespacesMap[namespace] = struct{}{}
+	}
+
+	var namespaces []string
+	for key := range namespacesMap {
+		namespaces = append(namespaces, key)
+	}
+
+	return namespaces
+}
+
+func getTerminationMessage(objectKind string, objects ...client.Object) string {
+	var objectFilteredNamespacedNames []types.NamespacedName
+	for _, obj := range objects {
+		if obj.GetObjectKind().GroupVersionKind().Kind == objectKind {
+			objectFilteredNamespacedNames = append(objectFilteredNamespacedNames, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		}
+	}
+
+	switch objectKind {
+	case virtv2.VirtualMachineKind:
+		return getVMTerminationMessage(objectFilteredNamespacedNames)
+	case virtv2.VirtualMachineBlockDeviceAttachmentKind:
+		return getVMBDATerminationMessage(objectFilteredNamespacedNames)
+	default:
+		return getDefaultTerminationMessage(objectKind, objectFilteredNamespacedNames)
+	}
+}
+
+func nameSpacedNamesToStringSlice(namespacedNames []types.NamespacedName) []string {
+	var result []string
+
+	for _, namespacedName := range namespacedNames {
+		if namespacedName.Namespace != "" {
+			result = append(result, fmt.Sprintf("%s/%s", namespacedName.Namespace, namespacedName.Name))
+		} else {
+			result = append(result, namespacedName.Name)
+		}
+	}
+
+	return result
+}
+
+func getDefaultTerminationMessage(objectKind string, objectNamespacedNames []types.NamespacedName) string {
+	switch len(objectNamespacedNames) {
+	case 1:
+		if objectNamespacedNames[0].Namespace != "" {
+			return fmt.Sprintf("the ClusterVirtualImage is currently being used to create the %s %s/%s", objectKind, objectNamespacedNames[0].Namespace, objectNamespacedNames[0].Name)
```

Comment:
```suggestion
			return fmt.Sprintf("the ClusterVirtualImage is in use by %s %s/%s", objectKind, objectNamespacedNames[0].Namespace, objectNamespacedNames[0].Name)
```

Creation may be finished already. "Is in use" is more clearer.